### ISSUE TITLE
Memory Management

### ISSUE THREAD

**Issue by Ai360n** [ContinualAI/avalanche-rl#21]
Hello,
In avalanhce-rl, after 1000 experiences and 1000 steps per experience, the below error will be risen. It is necessary to manage memory during training correctly.

torch.cuda.OutOfMemoryError
/Avalanche/avalanche-rl/avalanche_rl/training/strategies/rl_base_strategy.py", line 377, in train_exp
self.loss.backward()

Thanks.
